src/dmz_check.c

```diff
@@ -518,7 +518,11 @@ static int dmz_check_mapped_zone_bitmap(struct dmz_dev *dev,
 
 	/* No valid block should be present after the write pointer */
 	bad_bits = 0;
-	wp_block = dmz_sect2blk(zone->wp - zone->start);
+	/* Check for full zones */
+	if (zone->wp == (__u64)-1)
```

Comment:
I think it would be safer to use the zone condition here. The kernel normally sets the wp to start+len for full zones but I would not bet on that. ZBC defines the WP as invalid for full zones anyway, so any value could be in there (not necessarily -1, eventhough drives on the market tend to use that value).

Reply:
Right, will be doing so.
Was a bit unsure about the '-1' anyway.